### PR TITLE
Return multi errors in buildah-rm

### LINKED ISSUE #411: enhance error message when remove a container does not exist
`error reading build container: container not known` shows twice in the error message.

```
# buildah rm busybox-working-container-4
error reading build container "busybox-working-container-4": error reading build container: container not known
error reading build container: container not known
```

### PR DESCRIPTION
**Pull request by ripcurld0**
Fixes #411 

@gouyang can you cherry-pick and run tests to verify this?

Thanks.

### CODE REVIEW

**Comment by ripcurld0** [comment]
@TomSweeneyRedHat sure!

File: cmd/buildah/rm.go [hunk 1]
```diff
@@ -46,39 +55,27 @@ func rmCmd(c *cli.Context) error {
...
 		for _, builder := range builders {
 			id := builder.ContainerID
-			err = builder.Delete()
-			if e == nil {
-				e = err
-			}
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "error removing container %q: %v\n", builder.Container, err)
+			if err = builder.Delete(); err != nil {
+				lastError = writeError(os.Stderr, errors.Wrapf(err, "container %q", builder.Container), lastError)
```
Comment (rhatdan) [inline cmd/buildah/rm.go]: Shouldn't the wrapped error say
```
 +				lastError = writeError(os.Stderr, errors.Wrapf(err, "error removing container %q", builder.Container), lastError) 
```

File: cmd/buildah/rm.go [hunk 2]
```diff
@@ -46,39 +55,27 @@ func rmCmd(c *cli.Context) error {
...
 		}
 	} else {
 		for _, name := range args {
 			builder, err := openBuilder(store, name)
-			if e == nil {
-				e = err
-			}
 			if err != nil {
-				fmt.Fprintf(os.Stderr, "error reading build container %q: %v\n", name, err)
+				lastError = writeError(os.Stderr, errors.Wrapf(err, "container %q", name), lastError)
```
Comment (rhatdan) [inline cmd/buildah/rm.go]: ditto

File: cmd/buildah/rm.go [hunk 3]
```diff
@@ -46,39 +55,27 @@ func rmCmd(c *cli.Context) error {
...
-
 			id := builder.ContainerID
-			err = builder.Delete()
-			if e == nil {
-				e = err
-			}
-			if err != nil {
-				fmt.Fprintf(os.Stderr, "error removing container %q: %v\n", builder.Container, err)
+			if err = builder.Delete(); err != nil {
+				lastError = writeError(os.Stderr, errors.Wrapf(err, "container %q", name), lastError)
```
Comment (rhatdan) [inline cmd/buildah/rm.go]: ditto

File: cmd/buildah/rm.go (hunk 1 above)
Comment (ripcurld0) [reply cmd/buildah/rm.go]: 😞 my bad

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

File: cmd/buildah/rm.go [hunk 4]
```diff
@@ -27,7 +28,16 @@ var (
 	}
 )
 
+// writeError writes `lastError` into `w` if not nil and return the next error `err`
+func writeError(w io.Writer, err error, lastError error) error {
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/rm.go]: Just wondering if this might be better placed in util.go

File: tests/rm.bats [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+load helpers
+
+@test "remove multiple containers errors" {
+  run buildah --debug=false rm mycontainer1 mycontainer2 mycontainer3
+  [ "${lines[0]}" == "error removing container \"mycontainer1\": error reading build container: container not known" ]
+  [ "${lines[1]}" == "error removing container \"mycontainer2\": error reading build container: container not known" ]
+  [ "${lines[2]}" == "error removing container \"mycontainer3\": error reading build container: container not known" ]
+  [ $(wc -l <<< "$output") -eq 3 ]
+  [ "${status}" -eq 1 ]
```
Comment (TomSweeneyRedHat) [inline tests/rm.bats]: Thanks!

File: cmd/buildah/rm.go (hunk 4 above)
Comment (ripcurld0) [reply cmd/buildah/rm.go]: Right now, it's not used other than here.
I can look at `./cmd/buildah/rmi.go` (f.e.) and see if that can be used there too which makes it more reasonable to what you're suggesting.

does that sound OK?

**Comment by ripcurld0** [comment]
@TomSweeneyRedHat @rhatdan thanks

File: cmd/buildah/rm.go (hunk 4 above)
Comment (TomSweeneyRedHat) [reply cmd/buildah/rm.go]: 👍 That works for me @ripcurld0  Just keep it in the back of your mind and if you find a use for it somewhere else, moving it to util at that time would be fine.  Thanks!

**Pull request closed without merging**